gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/PipelineService.groovy

```diff
@@ -96,6 +97,27 @@ class PipelineService {
     orcaServiceSelector.withContext(RequestContext.get()).startPipeline(pipelineConfig, trigger.user?.toString())
   }
 
+  Map triggerViaEcho(String application, String pipelineNameOrId, Map parameters) {
+    def user = AuthenticatedRequest.getSpinnakerUser()
+    def eventId = UUID.randomUUID()
+    parameters.put("eventId", eventId)
+
+    Map eventMap = [
+      content: [
+        application: application,
+        pipelineNameOrId: pipelineNameOrId,
+        runAsUser: user.orElse(""),
```

Comment:
I don't know that we actually need `runAsUser` explicitly specified here? ... it should propagate automatically as the `X-SPINNAKER-USER` header and eventually make it's way to `orca`.

Reply:
Even then ... I would expect the `X-SPINNAKER-USER` header to be passed from `gate` -> `echo` (along with all `X-SPINNAKER-*`).

`echo` doesn't appear to be using the common https://github.com/spinnaker/kork/blob/master/kork-web/src/main/groovy/com/netflix/spinnaker/okhttp/SpinnakerRequestInterceptor.groovy which would explain why propagation isn't happening to orca.

```
retrofitBuilder.setRequestInterceptor(spinnakerRequestInterceptor)
```

It also looks like `echo` isn't using the common `AuthenticatedRequestFilter` which is responsible for extracting the headers and putting them on the MDC.

Looking deeper, it's actually important that we propagate `X-SPINNAKER-USER-ORIGIN` (not doing so would break our ability to shard executions that came in from `deck`). 

Would seem tidier to just make header handling and propagation consistent between echo and everything else. 

Happy to jump on a call tomorrow to make sure we're on the same page. 

If you'd like to get this merged, I'm familiar enough with the header propagation and happy to do a follow-up PR to make things consistent.

Reply:
We've also noticed that `echo` isn't using the common `OkHttpMetricsInterceptor`, @dreynaud has that on his list of things to fix up.

Reply:
Yeah, it probably makes sense to have a quick call today.  I'll send you a message on slack to set one up.

Reply:
I think what I’m getting hung up on is a desire to propagate these headers for the manual case to preserve existing behavior, but not necessarily for any onCompletion/Failure trigger.